eis_toolkit/statistical_analyses/statistical_tests.py

```diff
@@ -0,0 +1,123 @@
+import pandas as pd
+from beartype import beartype
+from beartype.typing import Literal, Optional, Sequence, Tuple
+from scipy.stats import chi2_contingency, shapiro
+
+from eis_toolkit import exceptions
+
+
+@beartype
+def check_empty_dataframe(data: pd.DataFrame):
+    """Check if the input dataframe is empty.
+
+    Args:
+        data: Input DataFrame
+
+    Raises:
+        EmptyDataFrameException: The input DataFrame is empty.
+    """
+    if data.empty:
+        raise exceptions.EmptyDataFrameException("The input DataFrame is empty.")
+
+
+@beartype
+def chi_square_test(data: pd.DataFrame, target_column: str) -> Sequence[Tuple[float, float, int]]:
+    """Compute Chi-square test for independence on categorical data.
+
+    Args:
+        data: DataFrame containing the input data.
+        target_column: Variable against which independence of other variables is tested.
+
+    Raises:
+        InvalidParameterValueException: The target_column is not in input DataFrame.
+
+    Returns:
+        Test statistics for each variable (except target_column).
+    """
+    check_empty_dataframe(data)
+
+    if target_column not in data.columns:
+        raise exceptions.InvalidParameterValueException("Target column not found in the DataFrame.")
+
+    statistics = []
+    for column in data.columns:
+        if column != target_column:
+            contingency_table = pd.crosstab(data[target_column], data[column])
+            chi_square, p_value, degrees_of_freedom, _ = chi2_contingency(contingency_table)
+            statistics.append((chi_square, p_value, degrees_of_freedom))
+
+    return statistics
+
+
+@beartype
+def normality_test(data: pd.DataFrame) -> Sequence[Tuple[float, float]]:
+    """Compute Shapiro-Wilk test for normality on numeric input data.
+
+    Args:
+        data: DataFrame containing the input data.
+
+    Returns:
+        Test statistics for each variable.
+    """
+    check_empty_dataframe(data)
+
+    statistics = []
+    for column in data.columns:
+        statistic, p_value = shapiro(data[column])
+        statistics.append((statistic, p_value))
+
+    return statistics
```

Comment:
Here also docstring can be a bit more detailed if possible. Also the same as above, so output as dictionary.